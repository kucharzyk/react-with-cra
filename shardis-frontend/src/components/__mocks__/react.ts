import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
