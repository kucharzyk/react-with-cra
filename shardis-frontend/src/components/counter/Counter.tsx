import * as React from 'react';
import { StatelessComponent } from 'react';
import Button from 'material-ui/Button';

export interface Props {
  count: number;
  onIncrement?: () => void;
  onDecrement?: () => void;
}

const Counter: StatelessComponent<Props> = (props) => (
  <div className="Counter">
    <h2>Count is: {props.count}</h2>
    <div>
      <Button fab={true} color={'secondary'} name="dec" onClick={props.onDecrement}>-</Button>
      <Button fab={true} color={'secondary'} name="inc" onClick={props.onIncrement}>+</Button>
    </div>
  </div>
);

Counter.defaultProps = {
  onIncrement: () => {
    // no action
  },
  onDecrement: () => {
    // no action
  },
};

export default Counter;
