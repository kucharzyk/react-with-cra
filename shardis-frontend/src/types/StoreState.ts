import { CounterState } from '../reducers/counter';

export default interface StoreState {
  counter: CounterState;
}
