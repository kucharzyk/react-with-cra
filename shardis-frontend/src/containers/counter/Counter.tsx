import { connect, Dispatch } from 'react-redux';
import StoreState from '../../types/StoreState';
import Counter from '../../components/counter/Counter';
import * as actions from '../../actions/';

export function mapStateToProps({counter}: StoreState) {
  return {
    count: counter.count
  };
}

export function mapDispatchToProps(dispatch: Dispatch<actions.CounterAction>) {
  return {
    onIncrement: () => {
      dispatch(actions.incrementCounter());
    },
    onDecrement: () => {
      dispatch(actions.decrementCounter());
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Counter);
