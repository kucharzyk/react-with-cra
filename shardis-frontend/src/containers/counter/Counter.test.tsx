import * as React from 'react';
import { StatelessComponent } from 'react';
import * as ReactDOM from 'react-dom';

import { create } from 'react-test-renderer';
import Counter from './Counter';
import StoreState from '../../types/StoreState';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';

describe('Counter', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {
    const mockStore = configureStore<StoreState>();
    const store = mockStore({
      counter: 1
    });

    WrappedComponent = () => (
      <Provider store={store}>
        <StaticRouter location={'/'} context={{}}>
          <Counter/>
        </StaticRouter>
      </Provider>
    );
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<WrappedComponent/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<WrappedComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
