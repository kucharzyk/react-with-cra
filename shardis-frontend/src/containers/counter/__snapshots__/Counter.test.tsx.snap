// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders snapshot 1`] = `
<div
  className="Counter"
>
  <h2>
    Count is: 
    1
  </h2>
  <div>
    <WithStyles(Button)
      color="accent"
      fab={true}
      name="dec"
      onClick={[Function]}
    >
      -
    </WithStyles(Button)>
    <WithStyles(Button)
      color="accent"
      fab={true}
      name="inc"
      onClick={[Function]}
    >
      +
    </WithStyles(Button)>
  </div>
</div>
`;
